src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,773 @@
+                    break;
+                case SystemObjectType sot:
+                    WriteSchemaObjectExampleValue(writer, sot, exampleValue, variableName);
+                    break;
+                case EnumType enumType:
+                    WriteEnumTypeExampleValue(writer, enumType, exampleValue);
+                    break;
+                default:
+                    WriteFrameworkTypeExampleValue(writer, cst, exampleValue, variableName);
+                    break;
+            }
+        }
+
+        public static string useVariableName(string variableName)
+        {
+            if (Utilities.StringExtensions.IsCSharpKeyword(variableName))
+            {
+                variableName = $"@{variableName}";
+            }
+            if (!variableNames.Contains(variableName))
+            {
+                variableNames.Add(variableName);
+                return variableName;
+            }
+
+            for (int i = 2; true; i++)
+            {
+                var newName = $"{variableName}{i}";
+                if (!variableNames.Contains(newName))
+                {
+                    variableNames.Add(newName);
+                    return newName;
+                }
+            }
+            throw new Exception($"Can't find a valid variable name start with {variableName}");
+        }
+
+        public static void clearVariableNames() {
+            variableNames = new HashSet<string>();
+        }
+
+        public string WriteExampleParameterDeclaration(CodeWriter writer, ExampleParameter exampleParameter, Parameter parameter)
+        {
+            var variableName = useVariableName(exampleParameter.Parameter.CSharpName());
+            writer.Append($"{parameter.Type} {variableName} = ");
+            WriteExampleValue(writer, parameter.Type, exampleParameter.ExampleValue, variableName);
+            writer.Line($";");
+
+            foreach (var tagLine in _tagsWriter.ToString(false).Split(Environment.NewLine))
+            {
+                writer.AppendRaw(tagLine);
+            }
+            _tagsWriter = new CodeWriter();
+            return variableName;
+        }
+
+        public static string BuildValueString(ExampleValue exampleValue, Schema schema)
+        {
+            return $"\"{exampleValue.RawValue}\"";
+        }
+
+        public override void Write()
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override void WriteSingletonResourceEntry(Resource resource, string singletonResourceIdSuffix)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override void WriteResourceCollectionEntry(Resource resource)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override ResourceType GetBranchResourceType(RequestPath branch)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
If we do not implement these methods in this class, please remove these overrides and make this class `abstract`